.github/workflows/ci-actions-incremental.yml

```
@@ -715,3 +778,9 @@ jobs:
         with:
           name: test-reports-native-windows-${{matrix.category}}
           path: 'test-reports.tgz'
+      - name: Upload Surefire reports (if build failed)
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: surefire-reports-native-windows-${{matrix.category}}
+          path: "**/target/surefire-reports/TEST-*.xml"
```

```suggestion
          path: "**/target/*-reports/TEST-*.xml"
```
to also catch failsafe reports. Although not all jobs execute have failsafe tests, it wouldn't hurt to use the broader pattern everywhere.

---

PS: this should work for Gradle:
```
**/build/test-results/test/TEST-*.xml
```

---

Ok, I generalized a bit but not exactly sure how things will go. We will have to experiment with it.